Userland/Applications/TextEditor/MainWidget.cpp

```diff
@@ -40,13 +40,16 @@
 #include <LibWeb/HTML/SyntaxHighlighter/SyntaxHighlighter.h>
 #include <LibWeb/OutOfProcessWebView.h>
 #include <Shell/SyntaxHighlighter.h>
+#include <fcntl.h>
 
 namespace TextEditor {
 
 MainWidget::MainWidget()
 {
     load_from_gml(text_editor_window_gml);
 
+    m_filesystem_client = FileSystemAccessClient::construct();
+
```

Comment:
```suggestion
MainWidget::MainWidget()
    : m_filesystem_client(FileSystemAccessClient::construct())
{
    load_from_gml(text_editor_window_gml);
```
This allows you to change `m_filesystem_client` to `NonnullRefPtr<FileSystemAccessClient>`.